### ISSUE TITLE
Validation allows CORS allowOriginRegex: '*' which is invalid 

### ISSUE THREAD

**Issue by bdecoste** [solo-io/gloo#7524]
### Gloo Edge Version

1.12.x (latest stable)

### Kubernetes Version

None

### Describe the bug

Control Plane validation (i.e. webhook) allows CORS `allowOriginRegex: '*'` which results in an invalid route as the protocol is required. 

### Steps to reproduce the bug

1. Apply VirtualService with Route with CORS `allowOriginRegex: '*'`
2. The VirtualService will be accepted but the VirtualService will not be accepted and flagged by `glooctl check` 

### Expected Behavior

Validation rejected the invalid `allowOriginRegex: '*'` config

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1206768562311555/1207815121171357) by [Unito](https://www.unito.io)

**Comment by afro-coder** [comment]
The logs show the following errors
```
[2024-07-05 09:18:52.417][1][warning][config] [external/envoy/source/common/config/grpc_subscription_impl.cc:128] gRPC config for type.googleapis.com/envoy.config.route.v3.RouteConfiguration rejected: no argument for repetition operator: *
[2024-07-05 09:18:52.518][1][warning][config] [external/envoy/source/common/config/grpc_subscription_impl.cc:128] gRPC config for type.googleapis.com/envoy.config.route.v3.RouteConfiguration rejected: no argument for repetition operator: *
[2024-07-05 09:18:52.618][1][warning][config] [external/envoy/source/common/config/grpc_subscription_impl.cc:128] gRPC config for type.googleapis.com/envoy.config.route.v3.RouteConfiguration rejected: no argument for repetition operator: *
[2024-07-05 09:18:52.721][1][warning][config] [external/envoy/source/common/config/grpc_subscription_impl.cc:128] gRPC config for type.googleapis.com/envoy.config.route.v3.RouteConfiguration rejected: no argument for repetition operator: *
[2024-07-05 09:18:52.822][1][warning][config] [external/envoy/source/common/config/grpc_subscription_impl.cc:128] gRPC config for type.googleapis.com/envoy.config.route.v3.RouteConfiguration rejected: no argument for repetition operator: *
[2024-07-05 09:18:52.924][1][warning][config] [external/envoy/source/common/config/grpc_subscription_impl.cc:128] gRPC config for type.googleapis.com/envoy.config.route.v3.RouteConfiguration rejected: no argument for repetition operator: *
```

And the following is shown in glooctl check
```
glooctl check -n gloo-watch-test
Checking deployments... OK
Checking pods... Note: Unhandled pod condition PodReadyToStartContainersNote: Unhandled pod condition PodReadyToStartContainersNote: Unhandled pod condition PodReadyToStartContainersNote: Unhandled pod condition PodReadyToStartContainersOK
Checking upstreams... OK
Checking upstream groups... OK
Checking auth configs... OK
Checking rate limit configs... OK
Checking VirtualHostOptions... OK
Checking RouteOptions... OK
Checking secrets... OK
Checking virtual services... OK
Checking gateways... OK
Checking proxies... 1 Errors!
Gloo has detected that the data plane is out of sync. The following types of resources have not been accepted: [{resource="type.googleapis.com/envoy.config.route.v3.RouteConfiguration"}]. Gloo will not be able to process any other configuration updates until these errors are resolved.
Problem while checking for gloo xds errors
Error: 2 errors occurred:
	* An update to your gateway-proxy deployment was rejected due to schema/validation errors. The envoy_http_rds_update_rejected{envoy_rds_route_config="listener-__-8080-routes",envoy_http_conn_manager_prefix="http"} metric increased.
You may want to try using the `glooctl proxy logs` or `glooctl debug logs` commands.

	* Problem while checking for gloo xds errors
```

**Comment by AkshayAdsul** [comment]
^One more issue related to this.

**Comment by nfuden** [comment]
to be released in 1.16.12 1.17.0
